### PR TITLE
test(card): add refactored tests for card component FE-4524

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour
Refactor for Card component to use new cypress-component-react framework for testing.
I have added test.js and deleted steps.js and feature file.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [ ] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x]  Run npx cypress open-ct to check if there is newly added test.file
- [x]  Check if the card.test.js file passed
- [x]  Compare all tests were moved/refactored from .feature file -> .test.js file
- [x] Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

File: src/components/card/card.test.js [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+            View Stripe Dashboard
+          </Link>
+        </CardColumn>
+      </CardFooter>
+    </Card>
+  );
+};
+
+context("Tests for Card component", () => {
+  describe("Should check Card component properties", () => {
```
Comment (DlgSHi) [inline src/components/card/card.test.js:300]:
```suggestion
  describe("should check Card component properties", () => {
```

File: src/components/card/card.test.js [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+    it.each([
+      ["small", "24px"],
+      ["medium", "32px"],
+      ["large", "48px"],
+    ])(
+      "Should check %s spacing and padding for Card component ",
```
Comment (DlgSHi) [inline src/components/card/card.test.js:306]:
```suggestion
      "should check %s spacing and padding for Card component ",
```

File: src/components/card/card.test.js [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+      (spacing, paddings) => {
+        CypressMountWithProviders(<CardComponent spacing={spacing} />);
+        card()
+          .should("have.css", "padding-left", paddings)
+          .and("have.css", "padding-right", paddings);
+      }
+    );
+
+    it.each(testData)(
+      "Should check %s string value for Card component",
```
Comment (DlgSHi) [inline src/components/card/card.test.js:316]:
```suggestion
      "should check %s as children for Card component",
```

File: src/components/card/card.test.js [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+      (stringValue) => {
+        CypressMountWithProviders(<Card>{stringValue}</Card>);
+        card().should("have.text", stringValue);
+      }
+    );
+
+    it.each([250, 500, 650, 300])(
+      "Should check %s width for Card component",
```
Comment (DlgSHi) [inline src/components/card/card.test.js:324]:
```suggestion
      "should check %s width for Card component",
```

File: src/components/card/card.test.js [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+        </CardColumn>
+      </CardFooter>
+    </Card>
+  );
+};
+
+context("Tests for Card component", () => {
+  describe("Should check Card component properties", () => {
+    it.each([
+      ["small", "24px"],
```
Comment (DlgSHi) [inline src/components/card/card.test.js:302]: you can use here a `number`? as:
```suggestion
      ["small", 24],
```

File: src/components/card/card.test.js [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+        card().should("have.css", "width", `${width}px`);
+      }
+    );
+
+    it.each([
+      [1, 2, 2, 2],
+      [3, 1, 4, 0],
+      [4, 2, 2, 2],
+    ])(
+      "drag %s Card from the column to %s column",
```
Comment (DlgSHi) [inline src/components/card/card.test.js:336]:
```suggestion
      "drag %s Card item to the %s column",
```

File: src/components/card/card.test.js [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+          .children()
+          .should("have.length", lengthOfFirst + 1);
+
+        draggableContainer(2)
+          .children()
+          .should("have.length", lengthOfSecond + 1);
+      }
+    );
+
+    it("Should check dataRole for Card component", () => {
```
Comment (DlgSHi) [inline src/components/card/card.test.js:359]:
```suggestion
    it("should check dataRole for Card component", () => {
```

File: src/components/card/card.test.js [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+      CypressMountWithProviders(<CardComponent dataRole="cypress" />);
+      card().should("have.attr", "data-role", "cypress");
+    });
+
+    it("Should check interactive for Card component", () => {
```
Comment (DlgSHi) [inline src/components/card/card.test.js:364]:
```suggestion
    it("should check interactive for Card component", () => {
```

File: src/components/card/card.test.js [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+      CypressMountWithProviders(<CardComponent interactive />);
+      card().click("right", { force: true });
```
Comment (DlgSHi) [inline src/components/card/card.test.js:366]: do we need `{force:true}` here ?

File: src/components/card/card.test.js [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+        .and(
+          "have.css",
+          "box-shadow",
+          "rgba(0, 20, 30, 0.2) 0px 5px 5px 0px, rgba(0, 20, 30, 0.1) 0px 10px 10px 0px"
+        );
+    });
+
+    const textalign = ["center", "left", "right"];
+    it.each(textalign)(
+      "should check %s alignemnet for Card component",
```
Comment (DlgSHi) [inline src/components/card/card.test.js:379]:
```suggestion
      "should check %s alignment for Card component",
```

File: src/components/card/card.test.js [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+      (align) => {
+        CypressMountWithProviders(<CardTextAlignment align={align} />);
+        columnCard().should("have.css", "text-align", align);
+      }
+    );
+
+    it("Should check click action for Card component", () => {
```
Comment (DlgSHi) [inline src/components/card/card.test.js:386]:
```suggestion
    it("should check click action for Card component", () => {
```

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: src/components/card/card.test.js [hunk 12]
```diff
@@ -0,0 +1,385 @@
...
+import Box from "../box";
+import {
+  card,
+  draggableCard,
+  draggableContainer,
+  columnCard,
+} from "../../../cypress/locators/card/index";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const textalignment = ["center", "left", "right"];
```
Comment (DlgSHi) [inline src/components/card/card.test.js:24]:
```suggestion
const textAlignment = ["center", "left", "right"];
```

File: src/components/card/card.test.js [hunk 13]
```diff
@@ -0,0 +1,385 @@
+/* eslint-disable react/prop-types */
```
Comment (DlgSHi) [inline src/components/card/card.test.js:1]: do we need this comment here ?

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Pull request merged**